### PR TITLE
Add fabric on Android

### PR DESCRIPTION
**Pull request by okwasniewski**
Summary:
---------
 - Configure gradle files
 - Implement / extend the generated native interfaces
 - Ensure backwards compatibility by splitting implementations into two folders.

Used examples / references: 
- [React native slider implementation](https://github.dev/facebook/react-native/blob/8b174a57c8e732bbfe7c69e7a5f66f48cb300539/ReactAndroid/src/main/java/com/facebook/react/views/slider/ReactSliderManager.java#L144)

Test Plan:
----------
- Setup fabric example to test the app using RN 0.70 RC

## Test plan for RN <0.70:
 
1. Set `newArchEnabled` to true inside `example/android/gradle.properties`
2. Add this line in `example/android/app/src/main/jni/Android.mk` file.
```diff
+ include $(NODE_MODULES_DIR)/@react-native-community/slider/android/build/generated/source/codegen/jni/Android.mk
include $(CLEAR_VARS)
```
In the same file above, go to the `LOCAL_SHARED_LIBRARIES` setting and add the following line:
```diff
    libreact_codegen_rncore \
+   libreact_codegen_ReactSlider \
    libreact_debug \
```
3. Update `example/android/app/src/main/jni/MainComponentsRegistry.cpp`
```diff
#include <react/renderer/components/rncore/ComponentDescriptors.h>
+ #include <react/renderer/components/ReactSlider/ComponentDescriptors.h>

...

MainComponentsRegistry::sharedProviderRegistry() {
  auto providerRegistry = CoreComponentsRegistry::sharedProviderRegistry();

  // Custom Fabric Components go here. You can register custom
  // components coming from your App or from 3rd party libraries here.
  //
  // providerRegistry->add(concreteComponentDescriptorProvider<
  //        AocViewerComponentDescriptor>());
+ providerRegistry->add(concreteComponentDescriptorProvider<RNCSliderComponentDescriptor>());

  return providerRegistry;
}
```

Then, to launch the project run:
```
npm run install && npm run example-android
```

### CODE REVIEW

**Comment by okwasniewski** [comment]
This PR is part of https://github.com/callstack/react-native-slider/issues/380 - linking this to the issue.

File: package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java [hunk 1]
```diff
@@ -0,0 +1,155 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
```
Comment (BartoszKlonowski) [inline package/android/src/oldarch/java/com/reactnativecommunity/slider/ReactSliderManager.java:6]: Let's remove the that, I don't think it's valid anymore.

**Review by BartoszKlonowski — APPROVED** [review]
I left some very minor comments, but it's totally well done!
Thank you! 👍

File: package/android/build.gradle [hunk 2]
```diff
@@ -28,6 +36,17 @@ android {
...
     targetSdkVersion getExtOrIntegerDefault('targetSdkVersion')
+    buildConfigField "boolean", "IS_NEW_ARCHITECTURE_ENABLED", isNewArchitectureEnabled().toString()
+  }
+
+  sourceSets {
+    main {
+      if (isNewArchitectureEnabled()) {
+          java.srcDirs += ['src/newarch']
+      } else {
+          java.srcDirs += ['src/oldarch']
```
[the remark this answers is not in the document]
Comment (BartoszKlonowski) [reply package/android/build.gradle:47]: The more I analyse it, the more I think that there's no need to revisit it, let's just keep that.

File: package/src/Slider.js [hunk 3]
```diff
@@ -307,7 +308,6 @@ const SliderComponent = (
       disabled={_disabled}
       onStartShouldSetResponder={() => true}
       onResponderTerminationRequest={() => false}
-      accessibilityState={_accessibilityState}
```
[the remark this answers is not in the document]
Comment (BartoszKlonowski) [reply package/src/Slider.js:310]: https://github.com/facebook/react-native/pull/34287

[the remark this answers is not in the document]
Comment (thymikee) [reply package/src/Slider.js:310]: FYI I requested to cherry-pick that commit to 0.70 https://github.com/reactwg/react-native-releases/discussions/26#discussioncomment-3323938

**Pull request merged**